src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -1172,9 +1172,11 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
             responseRows.add(responseRow);
         }
         
-        responseRows.addAll(getRemainingMissingResponseRows(question, possibleGiversWithoutResponses, 
-                                                            possibleReceiversWithoutResponsesForGiver, 
-                                                            prevGiver, viewType));
+        if (!responses.isEmpty()) {
+            responseRows.addAll(getRemainingMissingResponseRows(question, possibleGiversWithoutResponses, 
+                                                                possibleReceiversWithoutResponsesForGiver, 
+                                                                prevGiver, viewType));
+        }
```

Comment:
The typical case is that one student submits responses to all questions and that slows down everything. I thought we are not showing empty rows unless the user clicks that option (which is yet to be added)?